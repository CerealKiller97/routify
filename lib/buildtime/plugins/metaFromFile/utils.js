import { existsSync } from 'fs'
import fse from 'fs-extra'
import { resolve } from 'path'

// todo should be used

/**
 * caches result of a function and returns cached value on subsequent calls
 * @param {function} fn
 * @param {string} path
 * @param {boolean=} refresh
 */
const cached = async (fn, path, refresh) => {
    if (!existsSync(path) || refresh) {
        let value = await fn()
        value = value instanceof Function ? value.toString() : JSON.stringify(value)
        fse.outputFileSync(path, `export default ${value}`)
        return eval(value)
    }
    return import('file:///' + resolve(path)).then(r => r.default)
}
