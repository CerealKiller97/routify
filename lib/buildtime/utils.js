import fse from 'fs-extra'
import { pathToFileURL, fileURLToPath } from 'url'
import { dirname } from 'path'

/**
 * saves value to file and returns a dynamic-import function that returns the value
 * @param {string} outputDir dynamic import will be relative to this path
 * @param {string} file file to save data to
 * @param {any} value JSON.stringifiable value
 */
export const writeDynamicImport = (outputDir, file, value) => {
    fse.outputFileSync(`${outputDir}/${file}`, `export default ${value}`)
    return `() => import('./${file}').then(r => r.default)::_EVAL`
}

export const createDirname = meta => dirname(fileURLToPath(meta.url))

export function hookHandler() {
    let callbacks = []
    const hook = cb => {
        callbacks.push(cb)
        return () => {
            const index = callbacks.indexOf(cb)
            callbacks.splice(index, 1)
        }
    }
    hook.callbacks = callbacks
    hook.callCallbacks = attr => hook.callbacks.forEach(cb => cb(attr))

    return hook
}

/**
 * JSON.stringify wrapper that leaves values ending in ::_EVAL unquoted
 * @param {any} obj
 * @returns {string}
 */
export const stringifyWithEscape = obj => {
    const str = JSON.stringify(obj, null, 2)
    return str.replace(
        /^( *)"(.+?)": "(.+)::_EVAL"(,?)$/gm,
        (_, spaces, key, value, comma) =>
            `${spaces}"${key}": ${JSON.parse(`"${value}"`)}${comma}`,
    )
}
