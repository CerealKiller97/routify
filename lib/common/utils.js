import '../../typedef.js'

/**
 * @template {{}} T
 * @template {{}} T2
 * @param {T} target
 * @param  {...T2} sources
 * @return {T&Partial<T2>} //jsdoc unaware of mutation - incorrectly wants partial T2
 */
export function deepAssign(target, ...sources) {
    for (const source of sources) {
        if (Array.isArray(source)) {
            target = Array.isArray(target) ? target : []
            target.push(...source)
        } else
            for (const key of Reflect.ownKeys(source)) {
                if ([source[key], target[key]].every(isObjectOrArray)) {
                    target[key] = deepAssign(target[key], source[key])
                } else target[key] = source[key]
            }
    }
    return target
}

const isObject = v =>
    v &&
    typeof v === 'object' &&
    !['Array', 'Date', 'Regexp'].includes(v.constructor?.name)

export const isObjectOrArray = v => Array.isArray(v) || isObject(v)

/**
 * checks for repeating patterns to prevent infinite loops
 */
class RepetitionChecker {
    history = []
    /**
     * Pushes an entry to history. If the entry already exists, it checks
     * if the history between previously added entry and current entry is a
     * repetition. If so, it returns the repetition.
     * @param {any} entry
     * @returns {false|any[]}
     */
    pushAndCheck = entry => {
        const { history } = this
        const prevIndexCursor = history.lastIndexOf(entry) + 1
        history.push(entry)
        if (prevIndexCursor) {
            const lastIndexCursor = history.length
            const sliceLength = lastIndexCursor - prevIndexCursor
            const slice1 = history.slice(prevIndexCursor - sliceLength, prevIndexCursor)
            const slice2 = history.slice(prevIndexCursor)
            if (JSON.stringify(slice1) === JSON.stringify(slice2)) return slice2
        }
        return false
    }
}

/**
 * @param {RoutifyPlugin[]} _plugins
 * @returns {RoutifyPlugin[]}
 */
export const normalizePlugins = plugins =>
    // clone, flatten and normalize
    plugins.flat().map(plugin => ({
        ...plugin,
        before: [].concat(plugin.before).filter(Boolean), //convert to, or keep as array
        after: [].concat(plugin.after).filter(Boolean), //convert to, or keep as array
    }))

/**
 * @param {RoutifyPlugin[]} _plugins
 * @returns {RoutifyPlugin[]}
 */
export function sortPlugins(plugins) {
    const repetitionChecker = new RepetitionChecker()

    const _sort = () => {
        plugins.some((plugin1, index1) =>
            plugins.some((plugin2, index2) => {
                const plugin2ShouldPrecede =
                    plugin1.after.includes(plugin2.name) ||
                    plugin2.before.includes(plugin1.name)
                const plugin2DoesPrecede = index2 < index1
                if (plugin2ShouldPrecede && !plugin2DoesPrecede) {
                    // move plugin2 to the index of plugin1
                    plugins.splice(index2, 1) //remove
                    plugins.splice(index1, 0, plugin2) //insert

                    // make sure we're not stuck in a loop
                    const loop = repetitionChecker.pushAndCheck(plugin2)
                    if (loop)
                        // todo create loopException
                        throw new Error(
                            'found infinite loop in plugins. Repeating pattern:\n' +
                                `${loop
                                    .map(
                                        p =>
                                            `${p.name} (${['before', 'after']
                                                .map(name => `${name}: ${p[name]}`)
                                                .join('; ')} )`,
                                    )
                                    .join('\n')
                                    .replace(/before: ;/g, '')
                                    .replace(/after:  /g, '')
                                    .replace(/\( \)/g, '')}`,
                        )
                    _sort()
                    return true
                }
            }),
        )
    }

    _sort()

    return plugins
}

/**
 * @template {Routify} T
 * @param {T} instance
 * @param {Object.<string, any>} routes
 */
export const mockRoutes = (instance, routes) => {
    const queue = [{ parent: instance.superNode, children: routes }]

    while (queue.length) {
        const job = queue.shift()
        Object.entries(job.children).forEach(([name, routes]) => {
            queue.push({
                parent: job.parent.createChild(name, ''),
                children: routes,
            })
        })
    }

    return instance
}
