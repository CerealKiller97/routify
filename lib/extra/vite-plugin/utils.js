export async function getSpank() {
    try {
        // todo, remove ts-ignore when moving vite plugin to separate package
        // @ts-ignore
        const spank = await import('spank')
        return spank
    } catch (err) {
        console.log('[Routify] Could not import "spank".')
        console.log('[Routify] If you have not installed spank, please run:')
        console.log('[Routify] npm install spank')
        throw err
    }
}

export const stripLogs = async (id, code) =>
    id.match(/routify3?\/lib/) &&
    code
        .replace(/\/\/ *routify-dev-only-start[\s\S]+?\/\/ *routify-dev-only-end/gim, '')
        .replace(/.+ \/\/ *routify-dev-only/gi, '')
