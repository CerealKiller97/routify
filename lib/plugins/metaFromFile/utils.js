import { existsSync, mkdirSync, writeFileSync } from 'fs'
import { dirname, resolve } from 'path'

/**
 * @param {any} value
 * @param {string} path
 */

const split = (value, path) => {
    value = value instanceof Function ? value.toString() : JSON.stringify(value)
    mkdirSync(dirname(path), { recursive: true })
    writeFileSync(path, `export default ${value}`)
}

/**
 * @param {function} fn
 * @param {string} path
 * @param {boolean=} refresh
 */
const cached = async (fn, path, refresh) => {
    if (!existsSync(path) || refresh) {
        let value = await fn()
        value =
            value instanceof Function ? value.toString() : JSON.stringify(value)
        writeFileSync(path, `export default ${value}`)
        return eval(value)
    }
    return import('file:///' + resolve(path)).then(r => r.default)
}

export { split, cached }
