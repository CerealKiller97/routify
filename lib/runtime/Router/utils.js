import { get, writable } from 'svelte/store'

export const createActiveUrlStore = router => {
    const { set: _set, subscribe } = writable({ url: null, mode: 'pushState' })
    const modes = ['pushState', 'replaceState', 'popState']
    const history = []

    /** @param {{url: string, mode: 'pushState'|'replaceState'|'popState', origin: 'internal'|'address'}} url */
    const set = async (url, mode, origin) => {
        if (!origin) throw new Error('url must have origin')
        if (!modes.includes(mode))
            throw new Error(
                'url.mode must be pushState, replaceState or popState',
            )
        const lastEntry = history[history.length - 1]
        if (lastEntry && lastEntry.url === url) return false

        const urlObj = { url, mode, origin }
        history.push(urlObj)

        const oldValue = get({ subscribe })

        // beforeUrlChange
        for (const hook of router.beforeUrlChange.hooks) {
            const res = await hook(urlObj, oldValue, router)
            if (!res) return false
        }

        router.log.debug('activeUrl.set', urlObj)
        _set(urlObj)

        // afterUrlChange
        router.afterUrlChange.hooks.forEach(hook =>
            hook(urlObj, oldValue, router),
        )
    }

    return {
        set,
        push: (url, origin) => set(url, 'pushState', origin),
        replace: (url, origin) => set(url, 'replaceState', origin),
        pop: (url, origin) => set(url, 'popState', origin),
        get: () => get({ subscribe }),
        subscribe,
    }
}
