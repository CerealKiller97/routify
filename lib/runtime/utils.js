import { writable } from 'svelte/store'
import '#typedef.js'

/**
 * writable with subscription hooks
 * @param {any} value
 */
export const writable2 = value => {
    let subscribers = []
    let { set, subscribe, update } = writable(value)

    const hooks = {
        onSub: () => {},
        onUnsub: () => {},
        onFirstSub: () => {},
        onLastUnsub: () => {},
    }

    const newSubscribe = (run, invalidator) => {
        // hooks
        hooks.onSub()
        if (!subscribers.length) hooks.onFirstSub()

        const unsub = subscribe(run, invalidator)
        subscribers.push(unsub)
        return () => {
            hooks.onUnsub()
            if (subscribers.length === 1) hooks.onLastUnsub()

            subscribers = subscribers.filter(_unsub => _unsub !== unsub)
            unsub()
        }
    }

    return {
        set,
        subscribe: newSubscribe,
        update,
        subscribers,
        hooks,
    }
}

export const isDescendantElem = parent => elem => {
    while ((elem = elem.parentNode)) if (elem === parent) return true
    return false
}

export const getUrlFromClick = event => {
    const el = event.target.closest('a')
    const href = el && el.href

    if (
        event.ctrlKey ||
        event.metaKey ||
        event.altKey ||
        event.shiftKey ||
        event.button ||
        (event.key && event.key !== 'Enter') ||
        event.defaultPrevented
    )
        return
    if (!href || el.target || el.host !== location.host) return

    const url = new URL(href)
    const relativeUrl = url.pathname + url.search + url.hash

    event.preventDefault()
    return relativeUrl
}

// TODO ADD TEST
/**
 *
 * @param {string} path
 * @param {Object.<string|number,string|number>} params
 * @param {function} queryHandler
 * @returns {string}
 */
export const pathAndParamsToUrl = (path, params = {}, queryHandler) => {
    Object.entries(params).forEach(([key, val]) => {
        if (path.includes(`[${key}]`)) {
            path = path.replace(`[${key}]`, val)
            delete params[key]
        }
    })

    return path + queryHandler(params)
}

export const fromEntries = iterable => {
    return [...iterable].reduce((obj, [key, val]) => {
        obj[key] = val
        return obj
    }, {})
}

/**
 * creates a hook collection
 * @example
 * const hooksCollection = createHook()
 * const unhookFn = hooksCollection(x => console.log('hello', x))
 * const unhookFn2 = hooksCollection(x => console.log('goodbye', x))
 *
 * // call hooks
 * hooksCollection.hooks.forEach(hook => hook('Jake'))
 * // logs "hello Jake" and "goodbye Jake"
 *
 * // unregister
 * unhookFn()
 * unhookFn2()
 */
export const createHook = () => {
    const hooks = []

    /**
     * @param {Function} hook
     * @returns {Function} unhook
     */
    const addHook = hook => {
        hooks.push(hook)
        const index = hooks.indexOf(hook)
        const unhook = () => hooks.splice(index, 1)
        return unhook
    }
    addHook.hooks = hooks

    return addHook
}

export const urlFromAddress = () =>
    (({ pathname, search, hash }) => pathname + search + hash)(window.location)

let _globalAutoIncrementerCount = {}
export const autoIncrementer = (
    storeObj = _globalAutoIncrementerCount,
    name = '__NA',
) => {
    storeObj[name] = storeObj[name] || 0
    storeObj[name]++
    return storeObj[name]
}
