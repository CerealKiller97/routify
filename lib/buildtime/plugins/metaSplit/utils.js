import { mkdirSync } from 'fs'
import { writeFileIfDifferentSync } from '#lib/buildtime/utils.js'
import { dirname, relative } from 'path'

/**
 * creates import function that returns a stringified function for JSON
 * @param {string} path
 * @param {string} importFrom
 * @returns {()=>Promise<any>}
 */
const createResolver = (path, importFrom) => {
    const relativePath = relative(importFrom, path).replace(/\\/g, '/')

    const resolver = () => import(path).then(r => r.default)

    resolver.toJSON = () =>
        `() => import('./${relativePath}').then(r => r.default)::_EVAL`
    return resolver
}

/**
 * curried func
 * @param {string} importFrom
 */
export const split =
    (importFrom = '.routify') =>
    /**
     * @param {any} value
     * @param {string} path
     */
    (value, path) => {
        value = value instanceof Function ? value.toString() : JSON.stringify(value)
        mkdirSync(dirname(path), { recursive: true })
        writeFileIfDifferentSync(path, `export default ${value}`)

        return createResolver(path, importFrom)
    }

const stringHashCode = str =>
    str
        .split('')
        .reduce((a, b) => {
            a = (a << 5) - a + b.charCodeAt(0)
            return a & a
        }, 0)
        .toString(36)

export const hashObj = val => {
    let str = JSON.stringify(val)
    str = str || (val.toJSON ? val.toJSON() : val.toString())
    const length = str.length
    const snippet = str
        .replace(/[^\w]/gm, '_')
        .replace(/\_+/gm, '_')
        .replace(/^\_/, '')
        .slice(0, 20)
    return snippet + '-' + length + '-' + stringHashCode(str)
}
