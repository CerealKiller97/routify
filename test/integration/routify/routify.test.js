import { suite } from 'uvu'
import * as assert from 'uvu/assert'
import { RoutifyBackend } from '../../../lib/RoutifyBackend.js'
import '../../../typedef.js'
import { createDirname } from '../../../common/utils.js'
import { resolve } from 'path'
import { readFileSync } from 'fs'
import fse from 'fs-extra'

const test = suite('routify')

const __dirname = createDirname(import.meta)

test('can run routify with bundled plugins', async () => {
    const instance = new RoutifyBackend({
        routifyDir: resolve(__dirname, '.routify'),
        filemapper: {
            routesDir: resolve(__dirname, 'example'),
        },
    })
    await instance.start()
    assertSnapshot(
        'routify',
        readFileSync(
            resolve(__dirname, '.routify', 'routes.default.js'),
            'utf-8',
        ),
        0,
    )
    assertSnapshot(
        'bundles',
        readFileSync(
            resolve(
                __dirname,
                '.routify',
                'bundles',
                '_default_admin-bundle.js',
            ),
            'utf-8',
        ),
        0,
    )
})

/*
    todo temporarily enabled
    Test disabled due to this issue: https://github.com/lukeed/uvu/issues/110
    When this is fixed or a workaround is found, please uncomment
*/
test.run()

function assertSnapshot(name, content, update) {
    const filepath = `${__dirname}/fixtures/${name}.js`
    if (update) fse.outputFileSync(filepath, content)
    const expect = readFileSync(filepath, 'utf-8')
    assert.equal(content, expect)
}
