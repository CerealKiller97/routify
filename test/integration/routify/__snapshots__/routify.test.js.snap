// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can run routify with bundled plugins: bundles 1`] = `
"export { default as _default_admin } from '../../../example/admin/_reset.svelte'
export { default as _default_admin_bundleme_svelte } from '../../../example/admin/bundleme.svelte'"
`;

exports[`can run routify with bundled plugins: routify 1`] = `
"import * as _default from '../../example/_module.svelte'
import * as _default_index_svelte from '../../example/index.svelte'
import * as _default_blog__slug__svelte from '../../example/blog/[slug].svelte'
import * as _default_blog_prefixed_slug__svelte from '../../example/blog/prefixed[slug].svelte'
import * as _default_no_layouts_nested_foo_svelte from '../../example/no-layouts/nested/foo.svelte'
import * as _default_no_layouts_nested_index_svelte from '../../example/no-layouts/nested/index.svelte'

export const routes = {
  \\"id\\": \\"_default\\",
  \\"component\\": _default,
  \\"meta\\": {},
  \\"rootName\\": \\"default\\",
  \\"file\\": {
    \\"path\\": \\"test/integration/routify/example/_module.svelte\\",
    \\"dir\\": \\"test/integration/routify/example\\",
    \\"base\\": \\"_module.svelte\\",
    \\"ext\\": \\".svelte\\",
    \\"name\\": \\"_module\\"
  },
  \\"routifyDir\\": import.meta.url,
  \\"children\\": [
    {
      \\"id\\": \\"_default_admin\\",
      \\"name\\": \\"admin\\",
      \\"component\\": import(\\"./bundles/_default_admin-bundle.js\\").then(r => r._default_admin),
      \\"meta\\": {
        \\"reset\\": {
          \\"value\\": true
        },
        \\"bundle\\": {
          \\"value\\": true
        }
      },
      \\"file\\": {
        \\"path\\": \\"test/integration/routify/example/admin/_reset.svelte\\",
        \\"dir\\": \\"test/integration/routify/example/admin\\",
        \\"base\\": \\"_reset.svelte\\",
        \\"ext\\": \\".svelte\\",
        \\"name\\": \\"_reset\\"
      },
      \\"children\\": [
        {
          \\"id\\": \\"_default_admin_bundleme_svelte\\",
          \\"name\\": \\"bundleme\\",
          \\"component\\": import(\\"./bundles/_default_admin-bundle.js\\").then(r => r._default_admin_bundleme_svelte),
          \\"meta\\": {},
          \\"file\\": {
            \\"path\\": \\"test/integration/routify/example/admin/bundleme.svelte\\",
            \\"dir\\": \\"test/integration/routify/example/admin\\",
            \\"base\\": \\"bundleme.svelte\\",
            \\"ext\\": \\".svelte\\",
            \\"name\\": \\"bundleme\\"
          },
          \\"children\\": []
        }
      ]
    },
    {
      \\"id\\": \\"_default_blog\\",
      \\"name\\": \\"blog\\",
      \\"component\\": false,
      \\"meta\\": {},
      \\"file\\": {
        \\"path\\": \\"test/integration/routify/example/blog\\",
        \\"dir\\": \\"test/integration/routify/example\\",
        \\"base\\": \\"blog\\",
        \\"ext\\": \\"\\",
        \\"name\\": \\"blog\\"
      },
      \\"children\\": [
        {
          \\"id\\": \\"_default_blog__slug__svelte\\",
          \\"name\\": \\"[slug]\\",
          \\"component\\": _default_blog__slug__svelte,
          \\"meta\\": {
            \\"dynamic\\": {
              \\"value\\": true
            }
          },
          \\"file\\": {
            \\"path\\": \\"test/integration/routify/example/blog/[slug].svelte\\",
            \\"dir\\": \\"test/integration/routify/example/blog\\",
            \\"base\\": \\"[slug].svelte\\",
            \\"ext\\": \\".svelte\\",
            \\"name\\": \\"[slug]\\"
          },
          \\"children\\": []
        },
        {
          \\"id\\": \\"_default_blog_prefixed_slug__svelte\\",
          \\"name\\": \\"prefixed[slug]\\",
          \\"component\\": _default_blog_prefixed_slug__svelte,
          \\"meta\\": {
            \\"dynamic\\": {
              \\"value\\": true
            }
          },
          \\"file\\": {
            \\"path\\": \\"test/integration/routify/example/blog/prefixed[slug].svelte\\",
            \\"dir\\": \\"test/integration/routify/example/blog\\",
            \\"base\\": \\"prefixed[slug].svelte\\",
            \\"ext\\": \\".svelte\\",
            \\"name\\": \\"prefixed[slug]\\"
          },
          \\"children\\": []
        }
      ]
    },
    {
      \\"id\\": \\"_default_index_svelte\\",
      \\"name\\": \\"index\\",
      \\"component\\": _default_index_svelte,
      \\"meta\\": {},
      \\"file\\": {
        \\"path\\": \\"test/integration/routify/example/index.svelte\\",
        \\"dir\\": \\"test/integration/routify/example\\",
        \\"base\\": \\"index.svelte\\",
        \\"ext\\": \\".svelte\\",
        \\"name\\": \\"index\\"
      },
      \\"children\\": []
    },
    {
      \\"id\\": \\"_default_no_layouts\\",
      \\"name\\": \\"no-layouts\\",
      \\"component\\": false,
      \\"meta\\": {},
      \\"file\\": {
        \\"path\\": \\"test/integration/routify/example/no-layouts\\",
        \\"dir\\": \\"test/integration/routify/example\\",
        \\"base\\": \\"no-layouts\\",
        \\"ext\\": \\"\\",
        \\"name\\": \\"no-layouts\\"
      },
      \\"children\\": [
        {
          \\"id\\": \\"_default_no_layouts_nested\\",
          \\"name\\": \\"nested\\",
          \\"component\\": false,
          \\"meta\\": {},
          \\"file\\": {
            \\"path\\": \\"test/integration/routify/example/no-layouts/nested\\",
            \\"dir\\": \\"test/integration/routify/example/no-layouts\\",
            \\"base\\": \\"nested\\",
            \\"ext\\": \\"\\",
            \\"name\\": \\"nested\\"
          },
          \\"children\\": [
            {
              \\"id\\": \\"_default_no_layouts_nested_foo_svelte\\",
              \\"name\\": \\"foo\\",
              \\"component\\": _default_no_layouts_nested_foo_svelte,
              \\"meta\\": {},
              \\"file\\": {
                \\"path\\": \\"test/integration/routify/example/no-layouts/nested/foo.svelte\\",
                \\"dir\\": \\"test/integration/routify/example/no-layouts/nested\\",
                \\"base\\": \\"foo.svelte\\",
                \\"ext\\": \\".svelte\\",
                \\"name\\": \\"foo\\"
              },
              \\"children\\": []
            },
            {
              \\"id\\": \\"_default_no_layouts_nested_index_svelte\\",
              \\"name\\": \\"index\\",
              \\"component\\": _default_no_layouts_nested_index_svelte,
              \\"meta\\": {},
              \\"file\\": {
                \\"path\\": \\"test/integration/routify/example/no-layouts/nested/index.svelte\\",
                \\"dir\\": \\"test/integration/routify/example/no-layouts/nested\\",
                \\"base\\": \\"index.svelte\\",
                \\"ext\\": \\".svelte\\",
                \\"name\\": \\"index\\"
              },
              \\"children\\": []
            }
          ]
        }
      ]
    }
  ]
}"
`;
